ui-components/src/commonMain/kotlin/org/jetbrains/kotlinconf/ui/components/NowButton.kt

```diff
@@ -0,0 +1,84 @@
+package org.jetbrains.kotlinconf.ui.components
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.animation.animateContentSize
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.heightIn
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material3.Icon
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.unit.dp
+import kotlinconfapp.ui_components.generated.resources.Res
+import kotlinconfapp.ui_components.generated.resources.arrow_down_16
+import org.jetbrains.compose.resources.painterResource
+import org.jetbrains.compose.ui.tooling.preview.Preview
+import org.jetbrains.kotlinconf.ui.theme.KotlinConfTheme
+import org.jetbrains.kotlinconf.ui.theme.PreviewHelper
+
+private val NowButtonShape = RoundedCornerShape(
+    topEndPercent = 50,
+    bottomEndPercent = 50,
+)
+
+@Composable
+fun NowButton(
+    enabled: Boolean,
+    onClick: () -> Unit,
+    modifier: Modifier = Modifier,
+) {
+    val textColor by animateColorAsState(
+        if (enabled) KotlinConfTheme.colors.primaryTextInverted
+        else KotlinConfTheme.colors.noteText
+    )
+    val background by animateColorAsState(
+        if (enabled) KotlinConfTheme.colors.primaryBackground
+        else KotlinConfTheme.colors.tileBackground
+    )
+
+    Row(
+        modifier = modifier
+            .clip(NowButtonShape)
+            .clickable(onClick = onClick, enabled = enabled)
+            .background(background)
+            .animateContentSize()
+            .width(72.dp)
+            .heightIn(min = 36.dp),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.spacedBy(2.dp, Alignment.CenterHorizontally),
+    ) {
+        StyledText(
+            text = "Now",
+            style = KotlinConfTheme.typography.text2,
+            color = textColor,
+        )
+
+        AnimatedVisibility(enabled) {
+            Icon(
+                // TODO review icon sizing later
```

Comment:
We'll need to take a look at the exact icon resources we have imported. We currently have them in 16, 24, 28 size variants, but I suspect we might not need all of these (as they're vectors anyway).

Reply:
It depends on the vector export properties (sometimes line width can scale badly)